### PR TITLE
Feature/49 implement new imap group of commands

### PR DESCRIPTION
**Pull request by przegwar**

### CODE REVIEW

File: G1ANT.Addon.Net/Commands/ImapCloseCommand.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿
```
Comment (danieladamkaiser) [inline G1ANT.Addon.Net/Commands/ImapCloseCommand.cs]: empty line

File: G1ANT.Addon.Net/Commands/ImapCloseCommand.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
```
Comment (danieladamkaiser) [inline G1ANT.Addon.Net/Commands/ImapCloseCommand.cs]: emptier lien

File: G1ANT.Addon.Net/Commands/ImapCloseCommand.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        public ImapCloseCommand(AbstractScripter scripter) : base(scripter)
+        { }
+
+        public void Execute(Arguments arguments)
+        {
+            ImapHelper.DisconnectClient();
+        }
+    }
+}
+
```
Comment (danieladamkaiser) [inline G1ANT.Addon.Net/Commands/ImapCloseCommand.cs]: bro wtf
empty line
plis fix

File: G1ANT.Addon.Net/Commands/ImapCloseCommand.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
```
Comment (danieladamkaiser) [inline G1ANT.Addon.Net/Commands/ImapCloseCommand.cs]: and more wtf

File: G1ANT.Addon.Net/Commands/ImapCloseCommand.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+﻿
+
+using G1ANT.Language;
```
Comment (danieladamkaiser) [inline G1ANT.Addon.Net/Commands/ImapCloseCommand.cs]: Move usings after license info

File: G1ANT.Addon.Net/Commands/ImapGetEmailsCommand.cs [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using G1ANT.Language;
+using G1ANT.Language.Models;
+using MailKit;
+using MailKit.Search;
+
+namespace G1ANT.Addon.Net
+{
+    [Command(Name = "imap.getmails", Tooltip = "This command uses the IMAP protocol to check an email inbox and allows the user to analyze their messages received within a specified time span, with the option to consider only unread messages and/or mark all of the checked ones as read")]
```
Comment (danieladamkaiser) [inline G1ANT.Addon.Net/Commands/ImapGetEmailsCommand.cs:21]: Random spelling of IMAP (upper, lower cases)- make it consistent in Tooltips

File: G1ANT.Addon.Net/Commands/ImapGetEmailsCommand.cs [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+            [Argument(Required = false, Tooltip = "Ending date for messages to be checked")]
+            public DateStructure ToDate { get; set; } = new DateStructure(DateTime.Now);
+
+            [Argument(Required = false, Tooltip = "If set to `true`, only unread messages will be checked")]
+            public BooleanStructure OnlyUnreadMessages { get; set; } = new BooleanStructure(false);
+
+            [Argument(Required = false, Tooltip = "Mark analyzed messages as read")]
+            public BooleanStructure MarkAsRead { get; set; } = new BooleanStructure(true);
+
+            [Argument(Required = false, Tooltip = "Name of a list variable where the returned mail variables will be stored")]
```
Comment (danieladamkaiser) [inline G1ANT.Addon.Net/Commands/ImapGetEmailsCommand.cs:47]: I believe all required tags are redundant, especially when you provide command with default value of argument

File: G1ANT.Addon.Net/Commands/ImapGetEmailsCommand.cs [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+            [Argument(Required = false, Tooltip = "Starting date for messages to be checked")]
+            public DateStructure SinceDate { get; set; }
+
+            [Argument(Required = false, Tooltip = "Ending date for messages to be checked")]
+            public DateStructure ToDate { get; set; } = new DateStructure(DateTime.Now);
+
+            [Argument(Required = false, Tooltip = "If set to `true`, only unread messages will be checked")]
+            public BooleanStructure OnlyUnreadMessages { get; set; } = new BooleanStructure(false);
+
+            [Argument(Required = false, Tooltip = "Mark analyzed messages as read")]
```
Comment (danieladamkaiser) [inline G1ANT.Addon.Net/Commands/ImapGetEmailsCommand.cs]: wtf are analyzed messages?

**Review by danieladamkaiser — CHANGES_REQUESTED** [review]

File: G1ANT.Addon.Net/ImapHelper.cs [hunk 9]
```diff
@@ -7,19 +7,41 @@ namespace G1ANT.Addon.Net
 {
     public static class ImapHelper
```
Comment (lfronczyk) [inline G1ANT.Addon.Net/ImapHelper.cs]: Make it non-static `ImapConnectionManager` or `ImapManager`. You can make it a singleton (blah).

File: G1ANT.Addon.Net/Commands/ImapReconnectCommand.cs [hunk 10]
```diff
@@ -0,0 +1,31 @@
+﻿
+
+using G1ANT.Language;
```
Comment (danieladamkaiser) [inline G1ANT.Addon.Net/Commands/ImapReconnectCommand.cs]: Empty lines
move usings after license info

**Review by danieladamkaiser — APPROVED** [review]

File: G1ANT.Addon.Net/ImapManager.cs [hunk 11]
```diff
@@ -0,0 +1,64 @@
+﻿using MailKit;
+using MailKit.Net.Imap;
+using System;
+using System.Net;
+
+namespace G1ANT.Addon.Net
+{
+    public sealed class ImapManager
+    {
+        private static ImapManager _instance = null;
```
Comment (lfronczyk) [inline G1ANT.Addon.Net/ImapManager.cs]:
```csharp
public static ImapManager Instance { get; } = new ImapManager()
```

**Review by lfronczyk — APPROVED** [review]

**Pull request merged**